source/common/upstream/original_dst_cluster.cc

```diff
@@ -39,6 +39,9 @@ HostConstSharedPtr OriginalDstCluster::LoadBalancer::chooseHost(LoadBalancerCont
         dst_host = connection->connectionInfoProvider().localAddress();
       }
     }
+    if (dst_host && port_override_ > 0) {
+      dst_host = Network::Utility::getAddressWithPort(*dst_host.get(), port_override_);
```

Comment:
if `getAddressWithPort` returns `null` then dst_host will be set to null as well I think - maybe want to only override if its non-null?

Reply:
If getAddressWithPort returns null then something has gone seriously wrong and I think we should fall through to line 86.